New directory for OpenTelemetry QuickStart (#4016)
GoogleCloudPlatform/python-docs-samples: trace/cloud-trace-demo-app-opentelemetry/README.md

#### Demo Requirements
If you are using Cloud Shell, skip to the next section.

1. Install gcloud <https://cloud.google.com/sdk/install>
2. Install kubectl <https://kubernetes.io/docs/tasks/tools/install-kubectl/>
3. Install docker <https://docs.docker.com/install/>

#### Google Container Registry Image Setup
If you are using the provided image, skip to the next section.

4. Get default project id and set environment variable:
    `PROJECT_ID=$(gcloud config get-value project)`
5. Build Image:
    `docker build -t gcr.io/${PROJECT-ID}/cloud-trace-demo .`
6. Upload Image to Container Registry:
    `gcloud docker -- push gcr.io/${PROJECT-ID}/cloud-trace-demo-test:v1`

#### Create a GKE cluster
7. Enable Google Cloud and set up region and zone.
    `gcloud init`
8. Enable the GKE API & billing:
    `gcloud services enable container.googleapis.com`
9. Create a GKE cluster named "demo":
    `gcloud container clusters create demo`

#### Send Requests to See Generated Traces

> 10. If you are using your own image, please change the image variable in the following files:

Reviewer: Also, I think step 10 could be moved to the image setup section (after step 6) so that you don't need the "If you are using your own image" part.

5. Build Image:
    `docker build -t gcr.io/${PROJECT-ID}/cloud-trace-demo .`
6. Upload Image to Container Registry:
    `gcloud docker -- push gcr.io/${PROJECT-ID}/cloud-trace-demo-test:v1`
    
    If you are using your own image, please change the image variable in the following files to
    gcr.io/${PROJECT-ID}/cloud-trace-demo-test:v1:
    * [YAML](./app/demo-service-a.yaml)
    * [template B](./app/demo-service-b.yaml.template)
    * [template C](./app/demo-service-c.yaml.template)
#### Create a GKE cluster
7. Enable Google Cloud and set up region and zone.
    `gcloud init`
8. Enable the GKE API & billing:
    `gcloud services enable container.googleapis.com`
9. Create a GKE cluster named "demo":
    `gcloud container clusters create demo`

#### Send Requests to See Generated Traces

10. Run setup.sh to apply the YAML files, which deploys all three services to GKE
    `./setup.sh`
11. Send request to Service C:

    `curl -w "\n" $(kubectl get svc cloud-trace-demo-c -ojsonpath='{.status.loadBalancer.ingress[0].ip}')`
12. Visit [Trace List](https://pantheon.corp.google.com/traces/list) to check traces generated.
    Click on any trace in the graph to see the Waterfall View.

>     ![Screenshot](./example.png)

Author: > The screenshot doesn't render since example.png is part of this PR, but once it's in will this render inline with the text? If so, it would probably look better on its own line.

Does not render in line with text but added a new line as looked better in formatting.

your own following steps 4 - 6.

[![Open Cloud Trace Demo APP in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=trace/cloud-trace-demo-app/README.md&amp;cloudshell_tutorial=trace/cloud-trace-demo-app/README.md)

#### Demo Requirements
If you are using Cloud Shell, skip to the next section.

1. Install gcloud <https://cloud.google.com/sdk/install>
2. Install kubectl <https://kubernetes.io/docs/tasks/tools/install-kubectl/>
3. Install docker <https://docs.docker.com/install/>

#### Google Container Registry Image Setup
If you are using the provided image, skip to the next section.

4. Get default project id and set environment variable:

    `PROJECT_ID=$(gcloud config get-value project)`
5. Build Image:

    `docker build -t gcr.io/${PROJECT-ID}/cloud-trace-demo .`
6. Upload Image to Container Registry:

    `gcloud docker -- push gcr.io/${PROJECT-ID}/cloud-trace-demo-test:v1`
    
7. Change the image variables of the following files to:
    

>     `gcr.ip/${PROJECT-ID}/cloud-trace-demo-test:v1`

Reviewer 2: ```suggestion
    `gcr.io/${PROJECT-ID}/cloud-trace-demo-test:v1`
```

    
    * [YAML](./app/demo-service-a.yaml)
    * [template B](./app/demo-service-b.yaml.template)
    * [template C](./app/demo-service-c.yaml.template)
    
#### Create a GKE cluster

8. Enable Google Cloud and set up region and zone.

    `gcloud init`
9. Enable the GKE API & billing:

    `gcloud services enable container.googleapis.com`
10. Create a GKE cluster named "demo":

    `gcloud container clusters create demo`

#### Send Requests to See Generated Traces

11. Run setup.sh to apply the YAML files, which deploys all three services to GKE:

    `./setup.sh`
12. Send request to Service C:

    `curl -w "\n" $(kubectl get svc cloud-trace-demo-c -ojsonpath='{.status.loadBalancer.ingress[0].ip}')`

> 12. Visit [Trace List](https://pantheon.corp.google.com/traces/list) to check traces generated.

Reviewer 2: ```suggestion
12. Visit [Trace List](https://console.cloud.google.com/traces/list) to check traces generated.
```

# cloud-trace-demo-app-opentelemetry

Open this demo app in [Google Cloud Shell](https://cloud.google.com/shell/docs/). This includes necessary tools.

[![Open Cloud Trace Demo APP in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=trace/cloud-trace-demo-app/README.md&amp;cloudshell_tutorial=trace/cloud-trace-demo-app/README.md)

#### Demo Requirements
If you are using Cloud Shell, skip to the next section.

1. Install gcloud <https://cloud.google.com/sdk/install>
2. Install kubectl <https://kubernetes.io/docs/tasks/tools/install-kubectl/>
3. Install docker <https://docs.docker.com/install/>

    
#### Create a GKE cluster

> 8. Enable Google Cloud and set up region and zone.

Reviewer: Why do the step numbers skip from 3 to 8?

    `gcloud init`
9. Enable the GKE API & billing:

    `gcloud services enable container.googleapis.com`
10. Create a GKE cluster named "cloud-trace-demo", replace you GCP zone with

> your gcp-zone closest in proximity to you:

Reviewer: Is there a page that lists the available gcp-zones? If we can link to it here then that would help the reader complete this step faster.

> 6. Create a GKE cluster named "cloud-trace-demo", replace you 
> [GCP Zone](https://cloud.google.com/compute/docs/regions-zones) with your gcp-zone closest in proximity to you:

Reviewer: ```suggestion
6. Create a GKE cluster named "cloud-trace-demo", replacing `your-gcp-zone` below with the 
[GCP Zone](https://cloud.google.com/compute/docs/regions-zones) closest in proximity to you:
```

(I might not have the right line wrapping in the suggestion, so check that too.)

     ```
     gcloud container clusters create cloud-trace-demo\`
     --num-nodes 1 \
     --enable-basic-auth \
     --issue-client-certificate \
     --zone your-gcp-zone
     ```
12. Verify that you have access to the cluster:
    
    `kubectl get nodes`

#### Deploy The Cloud Trace Demo App

13. Build and tag the docker image for demo app:

    `docker build -t gcr.io/${PROJECT_ID}/cloud-trace-demo:v1 .`

> 14. Deploy resource to the cluster.

Reviewer: ```suggestion
14. Deploy resource to the cluster:
```

    `kubectl apply -f deployment.yaml`
15. Track the status of the deployment:

    `kubectl get deployments`
    

>     Deployment is complete, when all of the available deployments are ready. 

Reviewer: ```suggestion
    Deployment is complete when all of the available deployments are ready. 
```

16. Run the following command to see the pods the deployment created:
    
    `kubectl get pods`  

#### Deploy The Cloud Trace Demo Service

> 17. Create the cloud trace demo service

Reviewer: ```suggestion
17. Create the cloud trace demo service:
```

    
    `kubectl apply -f service.yaml`
18. Get the services IP address by running the following command:
    
    `kubectl get services`

> 19. Send curl request to the EXTERNAL_IP, replcae EXTERNAL_IP with the external ip found

Reviewer: ```suggestion
19. Send a curl request to the EXTERNAL_IP, replacing `EXTERNAL_IP` with the external IP address found
```

> in the above command:

Reviewer: Instead of "in the above command" it would be better to be more specific and say "in step 18" (or whatever number that step gets after the update from another comment).
